GetResultsTests#testToAndFromXContentEmbedded fails reproducibly

**Build scan**:

Not available, only seen locally.

**Repro line**:

```./gradlew :server:test --tests "org.elasticsearch.index.get.GetResultTests.testToAndFromXContentEmbedded" -Dtests.seed=3205E115CC20EB17 -Dtests.locale=pl-PL -Dtests.timezone=EST -Druntime.java=14```

**Reproduces locally?**:

Yes, 100%.

**Applicable branches**:

`master` (a904c7014ebcaeecc0305c956d38a53c46808da5)

**Failure history**:

Not seen in CI

**Failure excerpt**:

```
org.elasticsearch.index.get.GetResultTests > testToAndFromXContentEmbedded FAILED
    java.lang.AssertionError: expected: org.elasticsearch.index.get.GetResult<{
      "_index" : null,
      "_id" : null,
      "_seq_no" : 6646957961986097284,
      "_primary_term" : 7,
      "_routing" : "zEne",
      "found" : true,
      "_source" : {
        "lARIVmJe" : -8033289809137462302,
        "gjHzQtFcg" : [
          {
            "gffbiDUCb" : "HSoJwgpi",
            "sIBprepwOS" : 0.4510828074110138,
            "LDaaGAMJEW" : "dsZUiS",
            "VhULAmxZo" : 0.24047737661153457,
            "cQMWfNK" : "WQIzxlgRO"
          }
        ]
      },
      "fields" : {
        "ogBeX" : [
          {
            "wMFIY" : true,
            "Aglsj" : "KzFHPZmZed",
            "JjSfA" : 2022698992
          }
        ],
        "HnuSo" : [
          [
            -994235075
          ]
        ],
        "GCVqkHMBYi" : [
          -9,
          57,
          37,
          -87
        ],
        "KroFx" : [
          [
            -1732302012,
            -1883312517,
            -1367228499
          ],
          [
            921807463
          ],
          [
            -1791885005,
            1461412019
          ]
        ],
        "TByBh" : [
          0.86527264,
          0.8043281
        ]
      }
    }> but was: org.elasticsearch.index.get.GetResult<{
      "_index" : null,
      "_id" : null,
      "_seq_no" : 6646957961986097284,
      "_primary_term" : 7,
      "_routing" : "zEne",
      "found" : true,
      "_source" : {
        "lARIVmJe" : -8033289809137462302,
        "gjHzQtFcg" : [
          {
            "gffbiDUCb" : "HSoJwgpi",
            "sIBprepwOS" : 0.4510828074110138,
            "LDaaGAMJEW" : "dsZUiS",
            "VhULAmxZo" : 0.24047737661153457,
            "cQMWfNK" : "WQIzxlgRO"
          }
        ]
      },
      "fields" : {
        "ogBeX" : [
          {
            "wMFIY" : true,
            "Aglsj" : "KzFHPZmZed",
            "JjSfA" : 2022698992
          }
        ],
        "HnuSo" : [
          [
            -994235075
          ]
        ],
        "GCVqkHMBYi" : [
          -9,
          57,
          37,
          -87
        ],
        "KroFx" : [
          [
            -1732302012,
            -1883312517,
            -1367228499
          ],
          [
            921807463
          ],
          [
            -1791885005,
            1461412019
          ]
        ],
        "TByBh" : [
          0.86527264,
          0.8043281
        ]
      }
    }>
        at __randomizedtesting.SeedInfo.seed([3205E115CC20EB17:626C2A5B9F6A3F6F]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.elasticsearch.index.get.GetResultTests.testToAndFromXContentEmbedded(GetResultTests.java:120)
```

In case it's not obvious, `0.86527264` is a `Float` in one object and a `Double` in the other 😬 .

Test was added for the HLRC in #22586 so I'm labelling this to match.

--- a/server/src/test/java/org/elasticsearch/index/get/GetResultTests.java
+++ b/server/src/test/java/org/elasticsearch/index/get/GetResultTests.java
@@ -97,7 +97,6 @@ public class GetResultTests extends ESTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/67009")
     public void testToAndFromXContentEmbedded() throws Exception {
         XContentType xContentType = randomFrom(XContentType.values());
         Tuple<GetResult, GetResult> tuple = randomGetResult(xContentType);

--- a/test/framework/src/main/java/org/elasticsearch/test/RandomObjects.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/RandomObjects.java
@@ -142,7 +142,7 @@ public final class RandomObjects {
             }
         }
         if (value instanceof Float) {
-            if (xContentType == XContentType.CBOR || xContentType == XContentType.SMILE) {
+            if (xContentType.canonical() == XContentType.CBOR || xContentType.canonical() == XContentType.SMILE) {
                 // with binary content types we pass back the object as is
                 return value;
             }

